big-o/time-and-space-complexity.md

```diff
@@ -0,0 +1,122 @@
+# Time and Space Complexity
+
+## Learning Goals
+
+- Define time and space complexity
+
+## Introduction
+
+Computers run our programs pretty quickly. For our projects, when we run the code or run the tests, it takes a fraction of a second, sometimes maybe one or two seconds. Our computers use a tiny bit of memory while running the program.
+
+However, those fractions of seconds add up. And what would happen if a computer did not have enough memory to hold everything in your code? In programming, we love to think in hypotheticals, and a hypothetical to constantly think about is, "What would happen if our data set was millions of items?"
```

Comment:
I appreciate this; I personally have a bad habit of only imagining Big O as "theoretical" consequences.

I'll update this soon to be less "this is all theory," and more "there are consequences!"

I will likely still keep the language of "imagine a million" since that ends up being:
1. a nice hyperbole that will stick out to students and get the point across
2. it's used as a thread throughout the text :P